lang/src/org/partiql/lang/eval/StrictnessContext.kt

```diff
@@ -0,0 +1,51 @@
+package org.partiql.lang.eval
+
+import org.partiql.lang.ast.MetaContainer
+import org.partiql.lang.errors.ErrorCode
+import org.partiql.lang.errors.PropertyValueMap
+
+
+internal class ErrorDetails(
+    val metas: MetaContainer,
+    val message: String,
+    val errorCode: ErrorCode,
+    val errorContext: PropertyValueMap? = null
+)
+
+internal class StrictnessContext(val strictnessMode: StrictnessMode, val valueFactory: ExprValueFactory) {
+
+    inline fun errorIf(
+        test: Boolean,
+        crossinline detailsConstructor: () -> ErrorDetails,
+        crossinline otherwise: () -> ExprValue
+    ): ExprValue =
+        when {
+            test ->
+                when(strictnessMode) {
+                    StrictnessMode.STANDARD ->
+                        with(detailsConstructor()) {
+                            throw EvaluationException(
+                                message = message,
+                                errorCode = errorCode,
+                                errorContext = errorContext,
+                                cause = null,
+                                internal = false)
```

Comment:
Yes, this *stongly* implies that exceptions marked as `internal` will *never* result in `MISSING`. We will instead throw them as per usual.